PixiEditor.ChangeableDocument/Changes/Change.cs

```diff
@@ -2,9 +2,29 @@
 
 internal abstract class Change : IDisposable
 {
+    /// <summary>
+    /// Check's if the layer is mergeable with the <paramref name="other"/> change. Returns false if not overridden
+    /// </summary>
+    /// <param name="other"></param>
+    /// <returns></returns>
     public virtual bool IsMergeableWith(Change other) => false;
+    /// <summary>
+    /// Initializes and validates the changes parameter. Called before Apply and ApplyTemporarily
+    /// </summary>
+    /// <param name="target">The document the change will be applied on</param>
+    /// <returns><see cref="Success"/> if all parameters are valid, otherwise <see cref="Error"/></returns>
     public abstract OneOf<Success, Error> InitializeAndValidate(Document target);
+    /// <summary>
+    /// Applies the change to the <paramref name="target"/>
+    /// </summary>
+    /// <param name="firstApply">True if this is the first time Apply was called, always true if ignoreInUndo is true</param>
```

Comment:
The "always true if ignoreInUndo is true" seems confusing. Like yeah, technically the change won't be applied again if ignoreInUndo is set to true, but like, it's only called once that's why it's always true, not because it will be true in the second call